### ISSUE TITLE
Configuring social login providers without an Amplify backend

### ISSUE THREAD

**Issue by Leedwon** [aws-amplify/amplify-flutter#5597]
We have successfully integrated existing AWS Cognito resources into our Flutter app without using Amplify CLI, following the guidance from the [official Amplify documentation](https://docs.amplify.aws/flutter/build-a-backend/auth/use-existing-cognito-resources/#use-auth-resources-without-an-amplify-backend). The email + password authentication flow is working as expected.

Now, we want to extend the functionality by integrating social login providers into our Flutter app. However, the current documentation only covers the most basic setup and lacks information on how to integrate social providers without using the Amplify CLI.

Could you point us to documentation or provide guidance on how to configure social login providers without relying on the Amplify CLI? Example of `amplifyconfiguration.dart` with social login providers integration would be great.

**Comment by Equartey** [comment]
Hi @Leedwon, this process isn't documented yet. It's similar to what you've already done, you need to populate the outputs file with the necessary keys for social login. 

Without knowing which social logins you want to include, here are two ideas:

1. Deploy a sample app using gen 2 with all the configuration you desire and then replace the config values with your existing backend. 
2. Work off this sample config I have for a Facebook login (maybe missing details depending on provider):
```json
  "auth": {
    "user_pool_id": "your_user_pool_id",
    "aws_region": "your_aws_region",
    "user_pool_client_id": "your_user_pool_client_id",
    "identity_pool_id": "your_identity_pool_id",
    "mfa_methods": [],
    "standard_required_attributes": [
      "email"
    ],
    "username_attributes": [
      "email"
    ],
    "user_verification_types": [
      "email"
    ],
    "mfa_configuration": "NONE",
    "password_policy": {
      "min_length": 8,
      "require_lowercase": true,
      "require_numbers": true,
      "require_symbols": true,
      "require_uppercase": true
    },
    "oauth": {
      "identity_providers": [
        "FACEBOOK"
      ],
      "redirect_sign_in_uri": [
        "myapp://"
      ],
      "redirect_sign_out_uri": [
        "myapp://"
      ],
      "response_type": "code",
      "scopes": [
        "phone",
        "email",
        "openid",
        "profile",
        "aws.cognito.signin.user.admin"
      ],
      "domain": "you_domain.com"
    },
    "unauthenticated_identities_enabled": true
  },
```

**Comment by Leedwon** [comment]
Hi @Equartey,

Thanks for your help! I ended up creating a dummy project using the CLI, which allowed me to tweak my `amplifyconfiguration.dart` file and get it working.

For future readers: we also found this [schema](https://github.com/aws-amplify/amplify-backend/blob/main/packages/client-config/src/client-config-schema/schema_v1.json), which may be useful for making manual config changes.

Looking forward to seeing this documented, but for now, this workaround does the job.
